Cannot create keystore in 6.8.16 docker images -  unrecognized algorithm name: PBEWithMD5AndDES

**Logstash information**:

Please include the following information:

1. Logstash version (e.g. `bin/logstash --version`) 6.8.16
2. Logstash installation source (e.g. built from source, with a package manager: DEB/RPM, expanded from tar or zip archive, docker) docker BC
3. How is Logstash being run (e.g. as a service/service manager: systemd, upstart, etc. Via command line, docker/kubernetes) docker compose.


**JVM** (e.g. `java -version`):
openjdk version "1.8.0_292"
OpenJDK Runtime Environment (build 1.8.0_292-b10)
OpenJDK 64-Bit Server VM (build 25.292-b10, mixed mode)

[ERROR] 2021-05-20 15:00:24.729 [main] secretstorecli - Failed to create Logstash keystore. {:cause=>java.security.KeyStoreException: Key protection  algorithm not found: java.security.NoSuchAlgorithmException: unrecognized algorithm name: PBEWithMD5AndDES, :backtrace=>["org.logstash.secret.store.backend.JavaKeyStore.create(org/logstash/secret/store/backend/JavaKeyStore.java:91)", "org.logstash.secret.store.backend.JavaKeyStore.create(org/logstash/secret/store/backend/JavaKeyStore.java:39)", "org.logstash.secret.store.SecretStoreFactory.doIt(org/logstash/secret/store/SecretStoreFactory.java:111)", "org.logstash.secret.store.SecretStoreFactory.create(org/logstash/secret/store/SecretStoreFactory.java:75)", "org.logstash.secret.cli.SecretStoreCli.deleteThenCreate(org/logstash/secret/cli/SecretStoreCli.java:188)", "org.logstash.secret.cli.SecretStoreCli.create(org/logstash/secret/cli/SecretStoreCli.java:179)", "org.logstash.secret.cli.SecretStoreCli.command(org/logstash/secret/cli/SecretStoreCli.java:67)", "java.lang.reflect.Method.invoke(java/lang/reflect/Method.java:498)", "org.jruby.javasupport.JavaMethod.invokeDirectWithExceptionHandling(org/jruby/javasupport/JavaMethod.java:425)", "org.jruby.javasupport.JavaMethod.invokeDirect(org/jruby/javasupport/JavaMethod.java:292)", "RUBY.<class:SecretStoreCli>(/usr/share/logstash/lib/secretstore/cli.rb:35)", "RUBY.<main>(/usr/share/logstash/lib/secretstore/cli.rb:16)", "org.jruby.Ruby.runInterpreter(org/jruby/Ruby.java:891)", "org.jruby.Ruby.runInterpreter(org/jruby/Ruby.java:895)", "org.jruby.Ruby.runNormally(org/jruby/Ruby.java:784)", "org.jruby.Ruby.runNormally(org/jruby/Ruby.java:797)", "org.jruby.Ruby.runFromMain(org/jruby/Ruby.java:609)", "org.jruby.Main.doRunFromMain(org/jruby/Main.java:415)", "org.jruby.Main.internalRun(org/jruby/Main.java:307)", "org.jruby.Main.run(org/jruby/Main.java:234)", "org.jruby.Main.main(org/jruby/Main.java:206)"]}


The previously bundled JDK version in the docker images is: 
openjdk version "1.8.0_262"
OpenJDK Runtime Environment (build 1.8.0_262-b10)
OpenJDK 64-Bit Server VM (build 25.262-b10, mixed mode)

And this works with all older versions.

--- a/docker/templates/Dockerfile.j2
+++ b/docker/templates/Dockerfile.j2
@@ -16,7 +16,10 @@ FROM centos:7
 
 # Install Java and the "which" command, which is needed by Logstash's shell
 # scripts.
-RUN yum update -y && yum install -y java-1.8.0-openjdk-devel which && \
+# NOTE: pinning jdk 1.8.0.282 as .292 has an issue creating keystores
+# Logstash issue: https://github.com/elastic/logstash/issues/12917
+# OpenJDK issue: https://bugs.openjdk.java.net/browse/JDK-8266261
+RUN yum update -y && yum install -y java-1.8.0-openjdk-devel-1.8.0.282.b08 java-1.8.0-openjdk-headless-1.8.0.282.b08 which && \
     yum clean all
 
 # Provide a non-root user to run the process.

